packages/scene-composer/src/components/SceneComposerInternal.spec.tsx

```diff
@@ -0,0 +1,220 @@
+    getSceneUri: () => Promise.resolve('file://test.json'),
+    getSceneUrl: () => Promise.resolve('file://test.json'),
+    getSceneObject: (uri: string) => Promise.resolve(str2ab(sceneContent)),
+  };
+}
+
+describe('SceneComposerInternal', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render correctly with an empty scene in editing mode', async () => {
+    let container;
+    act(() => {
+      container = renderer.create(
+        <SceneComposerInternal config={{ mode: 'Editing' }} sceneLoader={createSceneLoaderMock('')} />,
+      );
+    });
+
+    await new Promise((resolve) => setTimeout(resolve, 1));
+
+    // shows the scene hierarchy browser
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render correctly with an empty scene in viewing mode', async () => {
+    let container;
+    act(() => {
+      container = renderer.create(
+        <SceneComposerInternal config={{ mode: 'Viewing' }} sceneLoader={createSceneLoaderMock('')} />,
+      );
+    });
+
+    await new Promise((resolve) => setTimeout(resolve, 1));
+
+    // shows the scene hierarchy browser
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render correctly with a valid scene in editing mode', async () => {
+    let container;
+    act(() => {
+      container = renderer.create(
+        <SceneComposerInternal config={{ mode: 'Editing' }} sceneLoader={createSceneLoaderMock(testScenes.scene1)} />,
+      );
+    });
+
+    await new Promise((resolve) => setTimeout(resolve, 1));
+
+    // shows the scene hierarchy browser
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render warning when minor version is newer', async () => {
+    let container;
+    act(() => {
+      container = renderer.create(
+        <SceneComposerInternal
+          config={{ mode: 'Editing' }}
+          sceneLoader={createSceneLoaderMock(invalidTestScenes.unsupportedMinorVersionScene)}
+        />,
+      );
+    });
+
+    await new Promise((resolve) => setTimeout(resolve, 1));
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render error when major version is newer', async () => {
+    let container;
+    act(() => {
+      container = renderer.create(
+        <SceneComposerInternal
+          config={{ mode: 'Editing' }}
+          sceneLoader={createSceneLoaderMock(invalidTestScenes.unsupportedMajorVersion)}
+        />,
+      );
+    });
+
+    await new Promise((resolve) => setTimeout(resolve, 1));
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render error when specVersion is invalid', async () => {
+    let container;
+    act(() => {
+      container = renderer.create(
+        <SceneComposerInternal
+          config={{ mode: 'Editing' }}
+          sceneLoader={createSceneLoaderMock(invalidTestScenes.invalidSpecVersionScene)}
+        />,
+      );
+    });
+
+    await new Promise((resolve) => setTimeout(resolve, 1));
+
+    expect(container).toMatchSnapshot();
```

Comment:
instead of a bunch of snapshorts, can we instead assert that there is the error we expect?

Reply:
We've been trying to mock the child components that will change for unrelated changed, and they are working fine so far. so we can keep them like this for now and switch to assert on the expected output once we see issues